src/Paket.Core/Versioning/SemVer.fs

```diff
@@ -125,40 +160,83 @@ type SemVerInfo =
     
     member x.AsString
         with get() = x.ToString()
+        
+    member x.Equals(y) =
+        x.Major = y.Major && x.Minor = y.Minor && x.Patch = y.Patch && x.Build = y.Build && x.PreRelease = y.PreRelease
 
     override x.Equals(yobj) = 
         match yobj with
-        | :? SemVerInfo as y -> 
-            x.Major = y.Major && x.Minor = y.Minor && x.Patch = y.Patch && x.PreRelease = y.PreRelease && x.Build = y.Build && x.BuildMetaData = y.BuildMetaData 
+        | :? SemVerInfo as y -> x.Equals(y)
         | _ -> false
     
-    override x.GetHashCode() = hash (x.Minor, x.Minor, x.Patch, x.PreRelease, x.Build)
+    override x.GetHashCode() = hash (x.Minor, x.Minor, x.Patch, x.Build, x.PreRelease)
```

Comment:
why the change in order here?

Reply:
Build is a number (or simple string) and may be part of release versions like `1.2.3.4` so the comparison preference should be kept regardless if there is PreRelease present or not; plus, this way it's faster :)

Reply:
yes, correct again :)

Reply:
Is this deliberately not including x.Major in the hash (though it does include x.Minor twice)?

Reply:
oh my, that's a very bad typo, thanks for the catch :)